README.md

```diff
@@ -68,58 +64,332 @@ To build all examples via gradle, simply call:
 gradlew.bat build # windows
 ```
 
-Once the initial build has completes feel free to inspect and edit the project with the code editor of your choice.
- - The top repository provides `IntelliJ` configurations, 
- - the [mps](mps) sub-project can be opened using `MPS 2020.3.6`, and 
+Once the initial build has completes feel free to inspect the [use cases](/doc/README.md#use-cases) and start components as described for each use case.
+Alternatively, you can inspect and edit the project with the code editor of your choice.
+ - The top repository provides `IntelliJ` configurations,
+ - the [mps](mps) sub-project can be opened using `MPS 2020.3.6`, and
  - the [dashboard](spa-dashboard-angular) is a `WebStorm` project.
 
 
-## Components in this repository
+
+## System architecture and components in this repository
 
 This project allows you to run different use cases.
-Depending on the chosen use case, different components are used.
-This section gives a detailed overview and links for each of these components.
+Depending on the chosen use case, only a subset of the components in this repository are used.
+
+
+## Full System architecture
+
+The full architecture includes components for multiple use cases.
+One does not need all components to realize individual use cases.
+
+[<img src="/doc/images/system-diagram-full.svg" width=80% >](https://app.diagrams.net/#Hmodelix/modelix-samples/main/doc/images/system-diagram-full.svg)
+
+In the following a short overview is given on each component.
+
+
+
+  1. **The MPS Language**
+
+     The language (meta-model) and model used in all examples.
+
+     [See the 'Language' section in the MPS README.md](mps/README.md#language)
+
+
+  2. **Generated model API (domain API)**
+
+     An API based on the meta-model generated with the [api-gen](https://github.com/modelix/api-gen) plugin from modelix.
+
+     [See the 'Generated API' section in the MPS README.md](mps/README.md#generated-api)
+
+  3. **Domain-specific openAPI**
+
+     In this example project, an extra domain-specific API layer is added which is defined in the [openAPI](https://www.openapis.org/) [specification](/openapi).
+     This layer is meant educational as there are no noteworthy abstractions happen in this definition.
```

Comment:
```suggestion
     This layer is meant educational as no noteworthy abstractions from the language itself happen in this definition.
```

From a point of view of the microservice-like architecture of the example the definition still serves a purpose. It decouples the frontend from potential changes to the language.